appservice/src/SiteWrapper.ts

```diff
@@ -82,8 +91,46 @@ export class SiteWrapper {
             await client.webApps.updateConfiguration(this.resourceGroup, this.appName, config);
     }
 
+    public async getAppServicePlan(client: WebSiteManagementClient): Promise<AppServicePlan> {
+        return await client.appServicePlans.get(this.planResourceGroup, this.planName);
+    }
+
+    public async deleteSite(client: WebSiteManagementClient, outputChannel: vscode.OutputChannel): Promise<void> {
+        const confirmMessage: string = localize('deleteConfirmation', 'Are you sure you want to delete "{0}"?', this.appName);
+        if (await vscode.window.showWarningMessage(confirmMessage, this._yes) !== this._yes) {
+            return;
+        }
+
+        let plan: AppServicePlan | undefined;
+        let deletePlan: boolean = false;
+
+        if (!this.slotName) {
+            // API calls not necessary for deployment slots
+            plan = await this.getAppServicePlan(client);
+        }
+
+        if (!this.slotName && plan.numberOfSites < 2) {
+            const message: string = localize('deleteLastServicePlan', 'This is the last app in the App Service plan "{0}". Do you want to delete this App Service plan to prevent unexpected charges?', plan.name);
+            const input: string | undefined = await vscode.window.showWarningMessage(message, this._yes, this._no);
+            if (input === undefined) {
+                return;
+            } else {
+                deletePlan = input === this._yes;
+            }
+        }
+
+        outputChannel.show();
+        outputChannel.appendLine(localize('Deleting', 'Deleting "{0}"...', this.appName));
+        if (this.slotName) {
+            await client.webApps.deleteSlot(this.resourceGroup, this.name, this.slotName);
+        } else {
+            await client.webApps.deleteMethod(this.resourceGroup, this.name, { deleteEmptyServerFarm: deletePlan });
+        }
+        outputChannel.appendLine(localize('DeleteSucceeded', 'Successfully deleted "{0}".', this.appName));
+    }
+
     public async deployZip(fsPath: string, client: WebSiteManagementClient, outputChannel: vscode.OutputChannel): Promise<void> {
-        const yes: string = 'Yes';
+        const yes: string = this._yes;
```

Comment:
Why define a const for the string if we have a class property already defined for localization??

Reply:
Haha just a 'find and replace' gone wrong